src/parser/jobs/smn/modules/Summons.tsx

```diff
@@ -313,31 +313,57 @@ export class Summons extends Analyser {
 				key: summon.start,
 				title: {
 					content: <>
-						{this.parser.formatEpochTimestamp(summon.start)}: <ActionLink {...this.data.getAction(summon.data.demiSummon.action)} />
+						{<span style={{ width: '200px', display: 'inline-block' }}>{this.parser.formatEpochTimestamp(summon.start)}: <ActionLink {...this.data.getAction(summon.data.demiSummon.action)} /></span>}
+						{this.getEgiIcons(summon)}
 					</>,
 				},
-				content: {content: data.display},
+				content: { content: data.display },
 			},
 			hasError: data.hasError,
 		}
 	}
 
-	private buildWindowOutput(summon: HistoryEntry<SummonWindow>) {
-		const demiRow = (summon.data.demiSummon?.action === this.data.actions.SUMMON_BAHAMUT.id) ?
-			this.buildBahamutRow(summon) :
-			this.buildPhoenixRow(summon)
+	private getEgiIcons(summon: HistoryEntry<SummonWindow>) {
+		const egis = [summon.data.titanSummon, summon.data.ifritSummon, summon.data.garudaSummon]
+		egis.sort((a, b) => {
+			if (a === undefined) { return -1 }
+			if (b === undefined) { return 1 }
+			return (a.timestamp > b.timestamp) ? 1 : -1
+		})
+
+		const retval: JSX.Element[] = []
+		egis.forEach(egi => {
+			if (egi !== undefined) {
+				if (egi === summon.data.ifritSummon) {
+					retval.push(< > <ActionLink showName={false} action="SUMMON_IFRIT_II" />&nbsp;{this.printUsageTime(summon.data.ifritSummon)} </>)
+				} else if (egi === summon.data.garudaSummon) {
+					retval.push(<> <ActionLink showName={false} action="SUMMON_GARUDA_II" />&nbsp;{this.printUsageTime(summon.data.garudaSummon)} </>)
+				} else if (egi === summon.data.titanSummon) {
+					retval.push(<> <ActionLink showName={false} action="SUMMON_TITAN_II" />&nbsp;{this.printUsageTime(summon.data.titanSummon)} </>)
+				}
+			}
+		})
+		return retval
+	}
+
+	private getEgiRows(summon: HistoryEntry<SummonWindow>) {
```

Comment:
Please return the table within each row to the original fixed sort order.

I did consider sorting by egi cast time when we first implemented this table but decided against it.  Fixed order was chosen to make it easier to scan through the tables since Ifrit/Titan/Garuda will always be in the same location.